sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/expressions.scala

```diff
@@ -403,14 +404,14 @@ object SimplifyConditionals extends Rule[LogicalPlan] with PredicateHelper {
           e.copy(branches = newBranches)
         }
 
-      case e @ CaseWhen(branches, _) if branches.headOption.map(_._1) == Some(TrueLiteral) =>
+      case CaseWhen(branches, _) if branches.headOption.map(_._1).contains(TrueLiteral) =>
```

Comment:
Since we removed `Scala 2.10`, it seems to okay. However, if we revert this irrelevant change, this PR becomes neater (and easier for someone to backport this).

Reply:
Since it's not a bug fix, I guess it's unlikely someone will backport this :)

Reply:
Yes. In the community, it's not allowed for backport. I mean the others who want to have this.

Reply:
Eh, in any event, wouldn't it be better to revert this change back if there's any actual advantage against a unrelated style change?

Reply:
Normally, we avoid adding unneeded refactoring in such a PR. Please avoid it next time. Thanks!